### PR TITLE
norisk-client: init at 0.5.10

### LINKED ISSUE #348132: Package request: norisk-client
**Project description**
NoRisk Client is a Minecraft client made by a German YouTuber.

**Metadata**

* homepage URL: https://norisk.gg/
* source URL: https://github.com/NoRiskClient/noriskclient-launcher
* license: GPL-3.0
* platforms: linux

**_Side note_**
The project is currently in private beta. The launcher can be launched, but Minecraft itself cannot. It will be released to the public by the end of the year, unless your Minecraft account is whitelisted.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by genga898**
Closes #348132

A minecraft client built using tauri [Noriskclient-launcher](https://norisk.gg)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Atemu** [comment]
@ofBorg eval

@BachErik could you test this?

File: pkgs/by-name/no/noriskclient-launcher/package.nix [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    forceGitDeps = true;
+    hash = "sha256-Rw8iyvnG0tK5b/tJ5fg5aS7HIvFuDHFRuWSpzAEq8uA=";
+  };
+
+  forceGitDeps = true;
+
+  makeCacheWritable = true;
+
+  cargoLock = {
+    lockFile = "${src}/${cargoRoot}/Cargo.lock";
```
Comment (SuperSandro2000) [inline pkgs/by-name/no/noriskclient-launcher/package.nix:44]: This triggers IFD, please copy the lockfile to nixpkgs.

**Comment by BachErik** [comment]
> @ofborg eval
> 
> @BachErik could you test this?

What and how should I test it?

File: pkgs/by-name/no/noriskclient-launcher/package.nix [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  ];
+
+  meta = {
+    description = "A minecraft client application made using svelte + tauri";
+    homepage = "https://norisk.gg";
+    changelog = "https://github.com/NoRiskClient/noriskclient-launcher";
+    license = lib.licenses.gpl3Only;
+    maintainers = with lib.maintainers; [ genga898 ];
+    mainProgram = "noriskclient-launcher";
+    platforms = lib.platforms.all;
```
[the remark this answers is not in the document]
Comment (genga898) [reply pkgs/by-name/no/noriskclient-launcher/package.nix]: Done

**Comment by Atemu** [comment]
Easiest would probably be to use nixpkgs-review as described in the PR template above and then run the app to see whether it works as intended.

**Comment by BachErik** [comment]
I am doing this the first time, so I did:
```bash
bacherik@nixos  ~/Documents/nixpkgs   master  nixpkgs-review pr --print-result 349304
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/349304/head:refs/nixpkgs-review/1
$ git worktree add /home/bacherik/.cache/nixpkgs-review/pr-349304-1/nixpkgs 394fbdae02bdd42040715203423d0e0bac78e8a4
Preparing worktree (detached HEAD 394fbdae02bd)
Updating files: 100% (43722/43722), done.
HEAD is now at 394fbdae02bd xournalpp: 1.2.3 -> 1.2.4 (#350094)
$ nix-env --extra-experimental-features no-url-literals --option system x86_64-linux -f <nixpkgs> --nix-path nixpkgs=/home/bacherik/.cache/nixpkgs-review/pr-349304-1/nixpkgs nixpkgs-overlays=/tmp/tmpd_jmn7a9 -qaP --xml --out-path --show-trace --no-allow-import-from-derivation
$ git merge --no-commit --no-ff f4f40d5ebc40cd38526097280c3f839b91770bd7
Automatic merge went well; stopped before committing as requested
$ nix-env --extra-experimental-features no-url-literals --option system x86_64-linux -f <nixpkgs> --nix-path nixpkgs=/home/bacherik/.cache/nixpkgs-review/pr-349304-1/nixpkgs nixpkgs-overlays=/tmp/tmpd_jmn7a9 -qaP --xml --out-path --show-trace --no-allow-import-from-derivation --meta
--------- Impacted packages on 'x86_64-linux' ---------
1 package added:
noriskclient-launcher (init at 0.5.10)

warning: ignoring the client-specified setting 'system', because it is a restricted setting and you are not a trusted user

$ nix build --file /nix/store/02nc48fjcip9bchnga0dwkxrqkryq4k3-nixpkgs-review-2.12.0/lib/python3.12/site-packages/nixpkgs_review/nix/review-shell.nix --nix-path 'nixpkgs=/home/bacherik/.cache/nixpkgs-review/pr-349304-1/nixpkgs nixpkgs-overlays=/tmp/tmpd_jmn7a9' --extra-experimental-features 'nix-command no-url-literals' --no-link --keep-going --no-allow-import-from-derivation --option build-use-sandbox relaxed --argstr local-system x86_64-linux --argstr nixpkgs-path /home/bacherik/.cache/nixpkgs-review/pr-349304-1/nixpkgs --argstr nixpkgs-config-path /tmp/tmpbyz3g4q_.nix --argstr attrs-path /home/bacherik/.cache/nixpkgs-review/pr-349304-1/attrs.nix
warning: ignoring the client-specified setting 'sandbox', because it is a restricted setting and you are not a trusted user

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/349304

--------- Report for 'x86_64-linux' ---------
1 package built:
noriskclient-launcher

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349304`

---
### `x86_64-linux`
<details>
<summary>:white_check_mark: 1 package built:</summary>
<ul>
<li>noriskclient-launcher</li>
</ul>
</details>


$ /nix/store/fpxbwqdma9rln9b6c21pz22famcvwsip-nix-2.24.9/bin/nix-shell --argstr local-system x86_64-linux --argstr nixpkgs-path /home/bacherik/.cache/nixpkgs-review/pr-349304-1/nixpkgs --argstr nixpkgs-config-path /tmp/tmpbyz3g4q_.nix --argstr attrs-path /home/bacherik/.cache/nixpkgs-review/pr-349304-1/attrs.nix --nix-path 'nixpkgs=/home/bacherik/.cache/nixpkgs-review/pr-349304-1/nixpkgs nixpkgs-overlays=/tmp/tmpd_jmn7a9' /nix/store/02nc48fjcip9bchnga0dwkxrqkryq4k3-nixpkgs-review-2.12.0/lib/python3.12/site-packages/nixpkgs_review/nix/review-shell.nix

[nix-shell:~/.cache/nixpkgs-review/pr-349304-1]$
```

But I dont know how to execute the programm.

**Comment by genga898** [comment]
Check for a /result/bin folder that is generated after the build and then run the program by writing ./{program_name}

**Comment by BachErik** [comment]
1. I have pulled this pr branch
2. 
```bash
cd nixpkgs
nix-build -A noriskclient-launcher
```
And the following error occurred:
```bash
bacherik@nixos  ~/Documents/nixpkgs   norisk-client  nix-build -A noriskclient-launcher
error:
       … while calling the 'abort' builtin
         at /home/bacherik/Documents/nixpkgs/lib/customisation.nix:269:13:
          268|        # which is especially relevant with allowAliases = false
          269|        else abort "lib.customisation.callPackageWith: ${error}";
             |             ^
          270|

       error: evaluation aborted with the following error message: 'lib.customisation.callPackageWith: Function called without required argument "webkitgtk_4_0" at /home/bacherik/Documents/nixpkgs/pkgs/by-name/no/noriskclient-launcher/package.nix:17, did you mean "webkitgtk_4_1", "webkitgtk_5_0" or "webkitgtk_6_0"?'
 ✘ bacherik@nixos  ~/Documents/nixpkgs   norisk-client 
```


But a friend of mine also tested it, and it has worked for him.

**Comment by genga898** [comment]
@BachErik you can edit the webkitgtk dependency, remove the `_4_0` from it and try rebuilding locally, for me it works both ways.

**Comment by BachErik** [comment]
Thanks, now it builds.
When I start the application, I get a gray screen (not that what should normally happen).

image

**Comment by Atemu** [comment]
The "No such file or directory" is suspicious and usually caused by Nix packaging bugs.

Please also confirm whether webkitgtk_4_0 still exists on master.

**Comment by genga898** [comment]
@BachErik That's weird, it shows a window similar to the web app version

**Comment by genga898** [comment]
@Atemu there's no webkitgtk_4_0 in master, it is just plain webkitgtk

**Comment by Atemu** [comment]
Please rebase atop of master then and make that work.

**Review by Atemu — CHANGES_REQUESTED** [review]
Please rebase atop of the current master (at least a1071d00556d8b7615c30bd2435be6a36ccb8eda) and ensure it still works with the webkitgtk changes.

**Comment by Atemu** [comment]
Rebase, no merges.

**Comment by Atemu** [comment]
I don't know what you did but that's not a rebase; you somehow squashed the merge commit into your actual commit?

Anyhow, you accidentally mass-pinged a bunch of people, which are now subscribed
and getting notifications for everything in this pull request. Unfortunately, they
cannot be automatically unsubscribed from the issue (removing review request does not
unsubscribe), therefore development cannot continue in this pull request anymore.

Please open a new pull request with your changes, link back to this one and ping the
people actually involved in here over there.

**Pull request closed without merging**